### PR TITLE
Drop the dependency on kitchen.

### PR DESCRIPTION
**Pull request by bowlofeggs**
Bodhi had been using kitchen to do a few different things, all of
which were either not useful or silly:

0. Encode strs into strs.
1. Encode strs into bytes. This is silly because we can do that
   just as easily and more correctly with the standard library
   (more correctly because kitchen uses the replace method on
   errors - we would be much better off to know about the errors
   so we can fix them).
2. Iterate on iterables, or turn a non-iterable into a single item
   iterable.

Each of these were replaced by using the standard library.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

### CODE REVIEW

**Comment by bowlofeggs** [comment]
@hroncok I got you an early Christmas present.

File: bodhi/server/mail.py [hunk 1]
```diff
@@ -412,14 +401,14 @@ def send_mail(from_addr, to_addr, subject, body_text, headers=None):
     if to_addr in config.get('exclude_mail'):
         return
 
-    subject = to_bytes(subject)
-    body_text = to_bytes(body_text)
+    subject = subject.encode('utf-8')
+    body_text = body_text.encode('utf-8')
 
-    msg = [b'From: %s' % to_bytes(from_addr), b'To: %s' % to_bytes(to_addr)]
+    msg = [b'From: %s' % from_addr.encode('utf-8'), b'To: %s' % to_addr.encode('utf-8')]
```
Comment (hroncok) [inline bodhi/server/mail.py]:
```suggestion
    msg = [f'From: {from_addr}'.encode('utf-8'), f'To: {to_addr}'.encode('utf-8')]
```

Comment (hroncok) [reply bodhi/server/mail.py]: Also, it might be easier to encode the message once it's joined.

Comment (hroncok) [reply bodhi/server/mail.py]: Something like https://github.com/bowlofeggs/bodhi/pull/2

**Review by Zlopez — APPROVED** [review]
LGTM

**Comment by bowlofeggs** [comment]
@hroncok I am going to apply this patch on top of yours to fix a couple of tests that expected the old behavior:

```
diff --git a/bodhi/tests/server/consumers/test_composer.py b/bodhi/tests/server/consumers/test_composer.py
index 6394bb00..dbcb64db 100644
--- a/bodhi/tests/server/consumers/test_composer.py
+++ b/bodhi/tests/server/consumers/test_composer.py
@@ -559,7 +559,7 @@ References:
                                 mock.ANY)
         self.assertEqual(len(mail.mock_calls), 2)
         body = mail.mock_calls[1][1][2]
-        self.assertTrue(body.decode('utf-8').startswith(
+        self.assertTrue(body.startswith(
             ('From: updates@fedoraproject.org\r\nTo: %s\r\nX-Bodhi: fedoraproject.org\r\nSubject: '
              'Fedora 17 updates-testing report\r\n\r\nThe following builds have been pushed to '
              'Fedora 17 updates-testing\n\n    bodhi-2.0-1.fc17\n\nDetails about builds:\n\n\n====='
diff --git a/bodhi/tests/server/test_mail.py b/bodhi/tests/server/test_mail.py
index 9fbd6552..af510456 100644
--- a/bodhi/tests/server/test_mail.py
+++ b/bodhi/tests/server/test_mail.py
@@ -351,7 +351,7 @@ class Test_SendMail(unittest.TestCase):
         mail._send_mail('archer@spies.com', 'lana@spies.com', 'hi')
 
         SMTP.assert_called_once_with('smtp.fp.o')
-        smtp.sendmail.assert_called_once_with('archer@spies.com', ['lana@spies.com'], 'hi')
+        smtp.sendmail.assert_called_once_with('archer@spies.com', ['lana@spies.com'], b'hi')
         warning.assert_called_once_with(
             '"recipient refused" for \'lana@spies.com\', {}'.format(
                 repr(smtp.sendmail.side_effect)))
```

**Comment by bowlofeggs** [comment]
The integration tests failed due to hitting a CI timeout. I've opened https://github.com/fedora-infra/bodhi/pull/3155 to raise that timeout.

I ran the CI tests locally on this patch and they all pass, so I'm going to merge this.

**Comment by bowlofeggs** [comment]
This patch is planned for inclusion in the upcoming 4.0.0 release: #3221

**Comment by bowlofeggs** [comment]
Bodhi 4.0.0 beta is built and deployed to staging:

https://copr.fedorainfracloud.org/coprs/bowlofeggs/bodhi-pre-release
https://bodhi.stg.fedoraproject.org/

**Pull request merged**